### ISSUE TITLE
module not found: '@angular/router.ngfactory' when use 'tns run android --bundle --env.aot --env.uglify'

### ISSUE THREAD

**Issue by miroslavaivanova** [NativeScript/nativescript-dev-webpack#630]
## Tell us about the problem
1. tns create app --ng
2. npm i nativescript-dev-webpack --save-dev
3. npm i nativescript-dev-typescript --save-dev
4. tns run android --bundle --env.aot --env.uglify
5. try to make a change in app/item/item.service.ts
the following exception will throw:
```
ERROR in ./app.module.ngfactory.js
Module not found: Error: Can't resolve '@angular/router.ngfactory' in '/Users/user/angApp1/app'
 @ ./app.module.ngfactory.js 10:0-111 10:0-111 27:0-48 49:211-244
 @ ./main.aot.ts
```

## Local environment
- [x ] NativeScript CLI version : 4.2.0

## Project data

```
"dependencies": {
    "@angular/animations": "~6.1.0",
    "@angular/common": "~6.1.0",
    "@angular/compiler": "~6.1.0",
    "@angular/core": "~6.1.0",
    "@angular/forms": "~6.1.0",
    "@angular/http": "~6.1.0",
    "@angular/platform-browser": "~6.1.0",
    "@angular/platform-browser-dynamic": "~6.1.0",
    "@angular/router": "~6.1.0",
    "nativescript-angular": "~6.1.0",
    "nativescript-theme-core": "~1.0.4",
    "reflect-metadata": "~0.1.8",
    "rxjs": "^6.0.0",
    "tns-core-modules": "~4.2.0",
    "zone.js": "^0.8.26"
  },
  "devDependencies": {
    "@angular/compiler-cli": "~6.1.0",
    "@ngtools/webpack": "~6.1.0",
    "nativescript-dev-typescript": "^0.7.2",
    "nativescript-dev-webpack": "^0.15.1",
    "typescript": "~2.7.2"
  },

**Comment by NathanWalker** [comment]
@miroslavaivanova I am also seeing this? Seemed to show up ever since the latest .1 release?

**Comment by NathanWalker** [comment]
@sis0k0 any insight on this one? You may be doing your talk at ngTalks right this very minutes so whenever/if you see this later :)

**Comment by bradmartin** [comment]
Same 😢 - hit it initially. Then change a file and it's gone, then it pops back up again. I removed a comment from the `app.module.ngfactory.d.ts` that was at the top of the file and it seemed to be run webpack fine, then it would fail. However, I don't know enough about the NS+Webpack to know what's going on.

**Comment by darkguy2008** [comment]
This can also be replicated with a simple angular project with routing included. Run "ng serve --aot" (or edit your package.json) and modify a file and et voilá. Same crappy error, I have the entire afternoon trying to deal with this with no avail.

**Comment by vchimev** [comment]
Thanks, @darkguy2008!

Reported in https://github.com/angular/angular-cli/issues/11835. Downgrade to `@angular/compiler@6.1.1` as a workaround.

**Comment by NathanWalker** [comment]
Downgrade sort of works. Ends up resulting in every other file change actually triggering livesync so doesn't work quite as well as it did prior whatever has caused the whole world to break here.
If anyone has a dependency version set which works 100% please post here.

**Comment by NathanWalker** [comment]
@darkguy2008 @NickIliev we are still seeing every other file change triggering livesync with 6.1.1 pinned angular version, you all seeing the same? Do we know if that for sure is angular related?
Certainly the description in this issue is Angular related but downgrading angular and {N} livesync is still broken to only trigger change (in particular a .html or .css/.scss file change) on every other file change so:

first view file save trggers just fine... 2nd one fails... 3rd one wins... 4th one fails...5th one wins, etc.

**Comment by NathanWalker** [comment]
I can confirm that pinning `"@ngtools/webpack": "6.1.1",` does **not** fix anything. Still same issue - every **other** view/style file change triggers.

**Comment by NickIliev** [comment]
@NathanWalker the issue with the `Can't resolve '@angular/router.ngfactory'` error should be resolved by pinning **"@angular/compiler"** (note that this is not "@angular/compiler-cli" or "@ngtools/webpack" ) to version 6.1.1. Doing that the livesync is working as expected on my side and changes made in TS, CSS or HTML files are respected every time.

Here is my `package.json`
```
  "dependencies": {
    "@angular/animations": "~6.1.0",
    "@angular/common": "~6.1.0",
    "@angular/compiler": "6.1.1",
    "@angular/core": "~6.1.0",
    "@angular/forms": "~6.1.0",
    "@angular/http": "~6.1.0",
    "@angular/platform-browser": "~6.1.0",
    "@angular/platform-browser-dynamic": "~6.1.0",
    "@angular/router": "~6.1.0",
    "nativescript-angular": "~6.1.0",
    "nativescript-theme-core": "~1.0.4",
    "reflect-metadata": "~0.1.8",
    "rxjs": "^6.0.0",
    "tns-core-modules": "~4.2.0",
    "zone.js": "^0.8.26"
  },
  "devDependencies": {
    "@angular/compiler-cli": "~6.1.0",
    "@ngtools/webpack": "~6.1.0",
    "nativescript-dev-typescript": "~0.7.0",
    "nativescript-dev-webpack": "^0.15.1",
    "typescript": "~2.7.2"
  }
```

And [here is the test applicaiton](https://github.com/NickIliev/NS-Issues-2018-II/tree/master/nativescript-dev-webpack/issue-630) I've used to test the livesync.

Angular are releasing a fix for the `Can't resolve '@angular/router.ngfactory'` issue so we would retest with the new `@angular/compiler` once the fix is published.

If you are still experiencing troubles with livesync on every other time, please post on what device/platform you are hitting this one - I guess it would be a candidate for opening a new issue as it seems to be a different case not related to this one.

**Comment by NathanWalker** [comment]
We'll try this and report back today - thank you for that 👍

**Comment by NathanWalker** [comment]
Additionally @NickIliev I cloned your issue repo [here](https://github.com/NickIliev/NS-Issues-2018-II/tree/master/nativescript-dev-webpack/issue-630) and ran the project and can see exactly the same problem so it's broken there as well.

```
Nathans-MacBook-Pro:nativescript-plugin-seed nathan$ node -v
v10.2.1
Nathans-MacBook-Pro:nativescript-plugin-seed nathan$ npm -v
6.4.0
```

Repro:

* run project with: `tns run ios --emulator --bundle --env.aot`
* after it launches, just open `items.component.html` and change ActionBar title to something else and then save.
* console will show:

```
File change detected. Starting incremental webpack compilation...
Webpack compilation complete. Watching for file changes.
Webpack build done!
```

And nothing happens. Save again and it will livesync. Save again and same thing so exactly the problem I describe above.

**Comment by NathanWalker** [comment]
@darkguy2008 referring to 2 different issues here - I may need to break the issue I mention out as it's different than description here. Only commenting here to avoid creating another issue if in fact it's related to this. Refer to these comments specifically:
https://github.com/NativeScript/nativescript-dev-webpack/issues/630#issuecomment-415136793
https://github.com/NativeScript/nativescript-dev-webpack/issues/630#issuecomment-415528877

Once @NickIliev has moment to review latest comments we'll see if we need to break out a new issue or not. Seeming like a distinctly separate issue not related to angular but {N} livesync with node 10+ possibly.

**Comment by NathanWalker** [comment]
I went ahead and create separate issue to track what I'm referring to here to avoid confusion:
https://github.com/NativeScript/nativescript-dev-webpack/issues/641

**Comment by darkguy2008** [comment]
@NathanWalker alright, thanks for your input. Actually thay issue was fixed with @angular/compiler-cli 6.1.6 released today, so I'd say that part of the issue could be already solved.

**Comment by NickIliev** [comment]
@darkguy2008 thank you for the update!

@miroslavaivanova  @NathanWalker @bradmartin I can confirm that indeed the latest version of @angular/compiler-cli (6.1.6) has resolved the issue.

To update the projects to use the latest version of `@angular/compiler-cli`  delete the `node_modules` folder and run [the upgrade scripts](https://docs.nativescript.org/releases/upgrade-instructions#updating-webpack-version-and-configuration) and then rerun `npm i` (or just manually upgrade the complier-cli version)
